### PR TITLE
Enable step operators on kubeflow

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
- Fix airflow example and some airflow related bugs
- Pass correct custom executor operator in kubeflow entrypoint and airflow orchestrator
- Validate that the stack contains the required step operators
- 
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/step_operators/entrypoint.py [hunk 1]
```diff
@@ -54,14 +53,14 @@ def create_executor_class(
...
     input_spec = {}
-    for input_name, class_path in input_artifact_type_mapping.items():
-        artifact_class = source_utils.load_source_path_class(class_path)
-        if not issubclass(artifact_class, BaseArtifact):
-            raise RuntimeError(
-                f"Class `{artifact_class}` specified as artifact class for "
-                f"input '{input_name}' is not a ZenML BaseArtifact subclass."
-            )
-        input_spec[input_name] = artifact_class
+    # for input_name, class_path in input_artifact_type_mapping.items():
```
Comment (htahir1) [inline src/zenml/step_operators/entrypoint.py:56]: Should we remove these?

File: src/zenml/step_operators/step_executor_operator.py [hunk 2]
```diff
@@ -110,7 +116,7 @@ def _collect_requirements(
         # TODO: Find a nice way to set this if the running version of ZenML is
         #  not an official release (e.g. on a development branch)
         requirements.add(
-            "git+https://github.com/zenml-io/zenml.git@feature/ENG-640-training-resource"
+            "git+https://github.com/zenml-io/zenml.git@feature/enable-training-resource-on-airflow-kubeflow"
```
Comment (htahir1) [inline src/zenml/step_operators/step_executor_operator.py:119]: I guess this will be removed later?

**Review by htahir1 — CHANGES_REQUESTED** [review]
Ok did an initial sweep, not many things but nits

File: src/zenml/step_operators/step_executor_operator.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/step_operators/step_executor_operator.py:119]: Yep exactly

File: examples/airflow_local/run.py [hunk 3]
```diff
@@ -21,14 +19,13 @@
...
+    importer=dynamic_importer(),
+    normalizer=normalize_mnist(),
+    trainer=sklearn_trainer(),
+    evaluator=sklearn_evaluator(),
+)
 
-    # Run the new pipeline
-    scikit_p.run()
+# Run the new pipeline
+DAG = scikit_p.run()
```
Comment (bcdurak) [inline examples/airflow_local/run.py:31]: :)

**Review by bcdurak — APPROVED** [review]
LGTM!

File: src/zenml/step_operators/entrypoint.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/step_operators/entrypoint.py]: Absolutely, I removed the comment and added an explanation why we can use just BaseArtifacts here

**Pull request merged**